include/ck_tile/ops/gemm/kernel/gemm_matrix_type.hpp

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2024, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+namespace ck_tile {
+enum struct MatrixALayout
+{
+    MK, // Row-major layout for matrix A (default)
+    KM  // Column-major layout for matrix A
+};
+
+enum struct MatrixBLayout
+{
+    NK, // Row-major layout for matrix B (default)
+    KN  // Column-major layout for matrix B
+};
+
+enum struct MatrixCLayout
+{
+    MN, // Row-major layout for matrix C (default)
+    NM  // Column-major layout for matrix C
+};
```

Comment:
In old CK we just have `RowMajor` and `ColMajor` this simplified a lot of layout specific code. In this way we would have to make specialized conditions for each A/B/C marix.